### PR TITLE
added toast notification ui

### PR DESCRIPTION
**Pull request by DanutIlie**

### CODE REVIEW

File: src/components/transaction-toast/components/transaction-toast-content/transaction-toast-content.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    let iconHtml = null;
+    if (typeof icon === 'string') {
+      iconHtml = getIconHtmlFromIconLookup(icon);
+    }
+    if (icon instanceof HTMLElement) {
+      iconHtml = icon.outerHTML;
+    }
+    return (
+      <div class="content">
```
Comment (arhtudormorar) [inline src/components/transaction-toast/components/transaction-toast-content/transaction-toast-content.tsx:29]: checkout if there were any data testids in sdk-dapp

File: src/components/transaction-toast/components/transaction-toast-details/transaction-toast-details.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+@Component({
+  tag: 'transaction-toast-details',
+  styleUrl: 'transaction-toast-details.css',
+  shadow: true,
+})
+export class TransactionDetails {
+  @Prop() processedTransactionsStatus?: JSX.Element | string;
+  @Prop() transactions?: ITransaction[];
+  @Prop() transactionClass: string;
```
Comment (arhtudormorar) [inline src/components/transaction-toast/components/transaction-toast-details/transaction-toast-details.tsx:12]: add TODO: use State Tunnel

File: src/components/transaction-toast/components/transaction-toast-progress/transaction-toast-progress.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  tag: 'transaction-toast-progress',
+  styleUrl: 'transaction-toast-progress.css',
+  shadow: true,
+})
+export class ToastProgress {
+  @Prop() progressClass: string = 'transaction-toast-progress';
+  @Prop() currentRemaining?: number;
+
+  render() {
+    return this.currentRemaining ? (
```
Comment (arhtudormorar) [inline src/components/transaction-toast/components/transaction-toast-progress/transaction-toast-progress.tsx:13]: make early return

File: src/index.html [hunk 4]
```diff
@@ -1,34 +1,114 @@
...
         if (!ledgerConnectModal) {
           const newModal = document.createElement('ledger-connect-modal');
           document.body.appendChild(newModal);
-        } 
+        }
+      }
+
+      let notificationId = 0;
+
+      function showNotification() {
```
Comment (arhtudormorar) [inline src/index.html:37]: add TODO: move to separate files

File: src/utils/getIconHtml.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { library, icon, IconDefinition, Icon, IconName } from '@fortawesome/fontawesome-svg-core';
+import * as solidIcons from '@fortawesome/free-solid-svg-icons';
+
+export function getIconHtmlFromIconLookup(iconName: IconName | string): string | null {
```
Comment (arhtudormorar) [inline src/utils/getIconHtml.ts:4]: make 2 files each with the name of the function so it's easy to find

Comment (DanutIlie) [reply src/utils/getIconHtml.ts]: done

File: src/index.html (hunk 4 above)
Comment (DanutIlie) [reply src/index.html]: i moved it now

File: src/components/transaction-toast/components/transaction-toast-progress/transaction-toast-progress.tsx (hunk 3 above)
Comment (DanutIlie) [reply src/components/transaction-toast/components/transaction-toast-progress/transaction-toast-progress.tsx]: done

File: src/components/transaction-toast/components/transaction-toast-details/transaction-toast-details.tsx (hunk 2 above)
Comment (DanutIlie) [reply src/components/transaction-toast/components/transaction-toast-details/transaction-toast-details.tsx:12]: i added todo on toast level

File: src/components/transaction-toast/components/transaction-toast-content/transaction-toast-content.tsx (hunk 1 above)
Comment (DanutIlie) [reply src/components/transaction-toast/components/transaction-toast-content/transaction-toast-content.tsx]: done, testids added

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**